### ISSUE TITLE
Error parsing `target_abi` cfg expression

### ISSUE THREAD

**Issue by mattbonnell** [bazelbuild/rules_rust#1637]
WORKSPACE
```
#-- Rust begin --#
load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

# To find additional information on this release or newer ones visit:
# https://github.com/bazelbuild/rules_rust/releases
load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

http_archive(
    name = "rules_rust",
    sha256 = "a8f5fe529f066068d98a72fa2dd61dbb410ecc2b9a7bd3bd412f529eb0cc9cd8",
    urls = ["https://github.com/bazelbuild/rules_rust/releases/download/0.12.0/rules_rust-v0.12.0.tar.gz"],
)

load("@rules_rust//rust:repositories.bzl", "rules_rust_dependencies", "rust_register_toolchains")

rules_rust_dependencies()

rust_register_toolchains(
    edition = "2021",
    version = "1.65.0",
)

load("@rules_rust//crate_universe:repositories.bzl", "crate_universe_dependencies")

crate_universe_dependencies()

# From http://bazelbuild.github.io/rules_rust/crate_universe.html#direct-packages
load("@rules_rust//crate_universe:defs.bzl", "crate", "crates_repository", "render_config")
load("//3rdparty/rust:defs.bzl", "rust_external_packages")

crates_repository(
    name = "crate_index",
    cargo_lockfile = "//:Cargo.bazel.lock",
    isolated = False,
    lockfile = "//:cargo-bazel-lock.json",
    packages = rust_external_packages(),
    # Setting the default package name to `""` forces the use of the macros defined in this repository
    # to always use the root package when looking for dependencies or aliases. This should be considered
    # optional as the repository also exposes alises for easy access to all dependencies.
    render_config = render_config(
        default_package_name = "",
    ),
    rust_version = "1.65.0",
    supported_platform_triples = [
	"x86_64-apple-darwin",
	"x86_64-unknown-linux-gnu",
	"aarch64-apple-darwin",
	"aarch64-unknown-linux-gnu",
    ]
)

load("@crate_index//:defs.bzl", "crate_repositories")

crate_repositories()

# From http://bazelbuild.github.io/rules_rust/rust_analyzer.html
load("@rules_rust//tools/rust_analyzer:deps.bzl", "rust_analyzer_dependencies")

rust_analyzer_dependencies()

#-- Rust end --#
```

`rust_external_packages` returns a list of `crate.spec`.

The error occurs when I attempt to add
```
    "parquet": crate.spec(
    	version = "26.0.0",
	default_features = False,
    ),
```

to my `crate_index.packages`. After running `REPIN=1 bazel sync --only=crate_index`, I get
```
Error in fail: Command [/private/var/tmp/_bazel_mabonnell/03580042f6da2d1e10179fa042f28312/external/crate_index/cargo-bazel,
"generate", "--cargo-lockfile", /Users/mabonnell/redacted/Cargo.bazel.lock, "--config", /private/va
r/tmp/_bazel_mabonnell/03580042f6da2d1e10179fa042f28312/external/crate_index/cargo-bazel.json, "--splicing-manifest", /privat
e/var/tmp/_bazel_mabonnell/03580042f6da2d1e10179fa042f28312/external/crate_index/splicing_manifest.json, "--repository-dir",
/private/var/tmp/_bazel_mabonnell/03580042f6da2d1e10179fa042f28312/external/crate_index, "--cargo", /private/var/tmp/_bazel_m
abonnell/03580042f6da2d1e10179fa042f28312/external/rust_darwin_x86_64__x86_64-apple-darwin_tools/bin/cargo, "--rustc", /priva
te/var/tmp/_bazel_mabonnell/03580042f6da2d1e10179fa042f28312/external/rust_darwin_x86_64__x86_64-apple-darwin_tools/bin/rustc
, "--lockfile", /Users/mabonnell/redacted/cargo-bazel-lock.json, "--repin", "--metadata", /private/
var/tmp/_bazel_mabonnell/03580042f6da2d1e10179fa042f28312/external/crate_index/splicing-output/metadata.json] failed with exit code 1.
STDOUT ------------------------------------------------------------------------

STDERR ------------------------------------------------------------------------
Error: Failed to parse expression: 'cfg(any(target_arch = "wasm32", target_abi = "unknown"))'

Caused by:
    any(target_arch = "wasm32", target_abi = "unknown")
                                ^^^^^^^^^^ expected one of `target_arch`, `target_feature`, `target_os`, `target_family`, `ta
rget_env`, `target_endian`, `target_has_atomic`, `target_pointer_width`, `target_vendor` here
```

**Comment by mattbonnell** [comment]
Ok, no error for `parquet = 14.0.0`. Going to work my way up and identify the version that introduces the issue

**Comment by mattbonnell** [comment]
`parquet = 21.0.0` is where the trouble starts. diffing it against `20.0.0`...

**Comment by mattbonnell** [comment]
Ok, `arrow-21.0.0` -- which `parquet-21.0.0` depends on -- upgrades the `ahash` dependency from `0.7` to `0.8`.

The `0.8` dependency is resolved with the recently-released ['ahash-0.8.1'](https://github.com/tkaitchuck/aHash/releases/tag/v0.8.1).

Whose Cargo.toml contains the troublesome lines
https://github.com/tkaitchuck/aHash/blob/673c17c930ef201233c14ee81136bad447cc2a5c/Cargo.toml#L83-L87

Relevant issue: https://github.com/tkaitchuck/aHash/issues/143

**Comment by carols10cents** [comment]
This should be fixed by https://github.com/bazelbuild/rules_rust/pull/1636
